src/overlay/overlay.scss

```diff
@@ -0,0 +1,379 @@
+    }
+  }
+
+  &.Overlay-header--large {
+    .Overlay-headerContentWrap {
+      .Overlay-titleWrap {
+        gap: $spacer-2;
+
+        .Overlay-title {
+          font-size: $h3-size;
+        }
+
+        .Overlay-description {
+          font-size: $body-font-size;
+        }
+      }
+    }
+  }
+
+  .Overlay-headerContentWrap {
+    display: flex;
+    align-items: flex-start;
+    gap: $spacer-2;
+    padding: $spacer-2 $spacer-2 0 $spacer-2;
+
+    .Overlay-actionWrap {
+      display: flex;
+      flex-direction: row;
+      gap: $spacer-2;
+    }
+
+    .Overlay-titleWrap {
+      display: flex;
+      padding: ($spacer-2 * 0.75) 0 ($spacer-2 * 0.75) $spacer-2;
+      flex-direction: column;
+      flex-grow: 1;
+      gap: $spacer-1;
+
+      .Overlay-title {
+        margin: 0;
+        font-size: $body-font-size;
+        font-weight: $font-weight-bold;
+      }
+
+      .Overlay-description {
+        margin: 0;
+        font-size: $font-size-small;
+        font-weight: $font-weight-normal;
+        color: var(--color-fg-muted);
+      }
+    }
+  }
+}
+
+// generic body content slot
+.Overlay-body {
+  padding: $spacer-3;
+  padding-top: 0;
+  overflow-y: scroll;
+  scrollbar-width: thin;
+  font-size: $body-font-size;
+  flex-grow: 1;
+
+  &.Overlay-body--paddingCondensed {
+    padding: $spacer-2;
+    padding-top: 0;
+  }
+
+  &.Overlay-body--paddingNone {
+    padding: 0;
+  }
+}
+
+// generic footer slot
+.Overlay-footer {
+  z-index: 1;
+  display: flex;
+  padding: 0 $spacer-3 $spacer-3 $spacer-3;
+  flex-direction: row;
+  flex-shrink: 0;
+  flex-wrap: wrap;
+
+  &.Overlay-footer--divided {
+    padding-top: $spacer-3;
+    // stylelint-disable-next-line primer/box-shadow
+    box-shadow: inset 0 $border-width var(--color-border-default);
+  }
+
+  &.Overlay-footer--alignStart {
+    justify-content: flex-start;
+    gap: $spacer-2;
+  }
+
+  &.Overlay-footer--alignCenter {
+    justify-content: center;
+    gap: $spacer-2;
+  }
+
+  &.Overlay-footer--alignEnd {
+    justify-content: flex-end;
+    gap: $spacer-2;
+  }
+}
+
+// TODO: replace with refactored IconButton
+.Overlay-closeButton {
+  position: relative;
+  display: grid;
+  width: $spacer-5;
+  height: $spacer-5;
+  padding: 0;
+  color: var(--color-fg-muted);
+  cursor: pointer;
+  user-select: none;
+  background-color: transparent;
+  border: $border-width $border-style transparent;
+  border-radius: $border-radius;
+  transition: 0.2s cubic-bezier(0.3, 0, 0.5, 1);
+  transition-property: color, background-color, border-color;
+  place-content: center;
+  align-self: flex-start;
+  flex-shrink: 0;
+
+  &:hover,
+  &:focus {
+    background-color: var(--color-btn-hover-bg);
+    border: $border-width $border-style var(--color-btn-hover-bg);
+  }
+}
+
+// this needs to move to a design token and be inline with JS
+// position will depend on conditional positioning logic
+.Overlay-backdrop-offset--normal {
+  top: 4px;
+}
+
+@mixin Overlay-backdrop() {
```

Comment:
Should we use a `.Overlay-backdrop` class instead of a mixin? We would have to add two classes when used:

```diff
- class="Overlay-backdrop--center"
+ class="Overlay-backdrop Overlay-backdrop--center"
```

but we should be able to save some file size because these "base styles" are not repeated for every variation.

Reply:
I see what you're saying.. it's a little more complicated though because we also have `Overlay-backdrop--transparent()`. So instead of overriding styles in each backdrop class `.Overlay-backdrop--anchor` I extracted it into a mixin 🤔 I think if I went back to classes, we would have to include more conditional logic within each class which might add more complexity to the selectors.

I think this CSS is a little more complex than I would like it to be (using mixins so extensively) but the responsive behavior _is_ complex 😂